src/containers/NewTaskModal/NewTaskModal.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import './NewTaskModal.css';
+import { addCard, hideAddModal } from '../../actions';
+
+class NewTaskModal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      title: '',
+      priority: '',
+      createdBy: '',
+      assignedTo: ''
+    };
+
+    this.handleTitleOnChange = this.handleTitleOnChange.bind(this);
+    this.handlePriorityOnChange = this.handlePriorityOnChange.bind(this);
+    this.handleCreatedByOnChange = this.handleCreatedByOnChange.bind(this);
+    this.handleAssignedToOnChange = this.handleAssignedToOnChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleTitleOnChange(e) {
+    const value = e.target.value;
+    this.setState({
+      title: value
+    });
+  }
+
+  handlePriorityOnChange(e) {
+    const value = e.target.value;
+    this.setState({
+      priority: value
+    });
+  }
+
+  handleCreatedByOnChange(e) {
+    const value = e.target.value;
+    this.setState({
+      createdBy: value
+    });
+  }
+
+  handleAssignedToOnChange(e) {
+    const value = e.target.value;
+    this.setState({
+      assignedTo: value
+    });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    const { title, priority, createdBy, assignedTo } = this.state;
+
+    this.props.onAdd({ title, priority, createdBy, assignedTo });
+
+    this.setState({
+      title: '',
+      priority: '',
+      createdBy: '',
+      assignedTo: ''
+    });
+  }
+
+  render() {
+    return (
+      <div className={this.props.newTaskModalDisplay}>
```

Comment:
As per my comment about the action passing a CSS property or even the reducer setting it, the better practice would be to have the reducer know about state and your UI to determine what to do based on that state rather than passing down a CSS property.